src/ConditionsResolver/Resovler/ConditionalGroup.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System.Text.RegularExpressions;
+using Monai.Deploy.WorkflowManager.ConditionsResolver.Extensions;
+
+namespace Monai.Deploy.WorkflowManager.ConditionsResolver.Resolver
+{
+    public class ConditionalGroup
+    {
+        public Keyword Keyword { get; set; }
+
+        public Conditional? LeftConditional { get; set; } = null;
+
+        public Conditional? RightConditional { get; set; } = null;
+
+        public ConditionalGroup? LeftGroup { get; set; } = null;
+
+        public ConditionalGroup? RightGroup { get; set; } = null;
+
+        public bool LeftIsSet => LeftGroup is not null || LeftConditional is not null;
+
+        public bool RightIsSet => RightGroup is not null || RightConditional is not null;
+
+        public int GroupedLogical { get; set; } = 1;
+
+        public Regex FindAnds { get; } = new Regex(@"([\s]and[\s]|[\s]AND[\s]|[\s]And[\s])");
+
+        public Regex FindOrs { get; } = new Regex(@"([\s]or[\s]|[\s]OR[\s]|[\s]Or[\s])");
+
+        public Regex FindBrackets { get; } = new Regex(@"((?<!\[)\()");
+
+        public Regex FindCloseBrackets { get; } = new Regex(@"((?<!\[)\))");
+
+        private string[] ParseOrs(string input) => FindOrs.SplitOnce(input);
+
+        private string[] ParseAnds(string input) => FindAnds.SplitOnce(input);
+
+        public void Set(string left, string right, Keyword keyword)
+        {
+            Keyword = keyword;
+            if (!string.IsNullOrEmpty(left))
+            {
+                if (FindAnds.Matches(left).Any() || FindOrs.Matches(left).Any())
+                {
+                    LeftGroup = ConditionalGroup.Create(left, GroupedLogical);
+                }
+                else
+                {
+                    LeftConditional = Conditional.Create(left);
+                }
+            }
+            if (!string.IsNullOrEmpty(right))
+            {
+                if (FindAnds.Matches(right).Any() || FindOrs.Matches(right).Any() || FindBrackets.Matches(right).Any())
+                {
+                    RightGroup = ConditionalGroup.Create(right, GroupedLogical);
+                }
+                else
+                {
+                    RightConditional = Conditional.Create(right);
+                }
+            }
+        }
+
+        public void Parse(string input, int groupedLogicalParent = 0)
+        {
+            var foundOpenBrackets = FindBrackets.Matches(input);
```

Comment:
null check?

Reply:
Good catches, I've also added tests for this.